plugins/inputs/sflow/decoder_test.go

```diff
@@ -159,7 +202,7 @@ func TestExpandFlow(t *testing.T) {
 				"ip_total_length":    uint64(0x05a6),
 				"ip_ttl":             uint64(0x7e),
 				"sampling_rate":      uint64(0x0400),
-				"tcp_header_length":  uint64(0x40),
+				"tcp_header_length":  uint64(0x14),
```

Comment:
Can we try to cross referenced this change with host-sflow, sflowtool/wireshark?

Reply:
yeah. The relevant documentation is: 
> Data offset (4 bits)
> Specifies the size of the TCP header in 32-bit words. The minimum size header is 5 words and the maximum is 15 words thus giving the minimum size of 20 bytes and maximum of 60 bytes, allowing for up to 40 bytes of options in the header. This field gets its name from the fact that it is also the offset from the start of the TCP segment to the actual data.

from https://en.wikipedia.org/wiki/Transmission_Control_Protocol

It's only 4 bits. thus the max is 1111(binary) = 16(dec). 15 (the max) * 4-byte-words = 60 bytes max value for this field. given that the test was expecting 0x40, which is 64(dec), there's no way the test was correct. The original code misread the 4 byte mask 1111 as 11110000, making the value 0101 (0x05) appear as if it was 01010000 (0x50). After multiplying by 4, the correct answer is 00010100 (0x14), but they got 0x0140, which overflowed the one byte value, dropping the first bit and resulting in 0x40 (again, larger than the max value allowed for this field). So it was definitely an implementation error.